tests/test_branch.py

```diff
@@ -51,7 +51,7 @@ def test_branches(
         r = cli_runner.invoke(["checkout", "HEAD~1"])
         assert r.exit_code == 0, r
 
-        assert text_branches(cli_runner) == ["* (no branch)", "  master"]
+        assert text_branches(cli_runner) == ["* (HEAD detached at 63a9492)", "  master"]
```

Comment:
so this changes because git reads it back from the reflog? 🤯 👍

Reply:
it's inexplicable to me, but yes git seems to do this using the reflog... somehow.